### PR TITLE
Task 53 implement code snippets

### PR DESCRIPTION
**Pull request by NarcisNacev1**
i taka

### CODE REVIEW

File: user/serializers.py [hunk 1]
```diff
@@ -84,3 +74,26 @@ def validate(self, attrs):
         data = super().validate(attrs)
         data['user'] = LoginProfileSerializer(self.user).data
         return data
+
+class CodeSnippetSerializer(serializers.ModelSerializer):
+    created_by = serializers.ReadOnlyField(source='created_by.username')
+    created_at = serializers.ReadOnlyField()
```
Comment (themane04) [inline user/serializers.py:80]: Use ShortUserSerializer

File: user/views.py [hunk 2]
```diff
@@ -1,11 +1,17 @@
...
+from rest_framework.permissions import AllowAny, IsAuthenticatedOrReadOnly
 from rest_framework.response import Response
 from rest_framework import viewsets
 from rest_framework_simplejwt.views import TokenObtainPairView
+from urllib3 import request
 
 from user.serializers import UserSerializer, CustomTokenObtainPairSerializer
 
+from rest_framework.decorators import action
+from django.shortcuts import get_object_or_404
```
Comment (themane04) [inline user/views.py:11]: from rest_framework

File: user/views.py [hunk 3]
```diff
@@ -27,3 +33,66 @@ class CustomTokenObtainPairView(TokenObtainPairView):
...
+class CodeSnippetViewSet(viewsets.ViewSet):
+    """
+    A viewset for viewing, editing, creating, and deleting code snippets.
+    """
+    permission_classes = [IsAuthenticatedOrReadOnly]
+
+    def list(self, request):
+        """
+        Retrieve a list of code snippets.
+        """
```
Comment (themane04) [inline user/views.py:47]: Comments only for the View and not for each method.

File: user/serializers.py [hunk 4]
```diff
@@ -84,3 +74,26 @@ def validate(self, attrs):
...
+        return CodeSnippet.objects.create(created_by=user, **validated_data)
+
+    def update(self, instance, validated_data):
+        instance.title = validated_data.get('title', instance.title)
+        instance.code = validated_data.get('code', instance.code)
+        instance.language = validated_data.get('language', instance.language)
+        if 'image' in validated_data:
+            instance.image = validated_data['image']
+        instance.save()
+        return instance
```
[the remark this answers is not in the document]
Comment (NarcisNacev1) [reply user/serializers.py:99]: ti si vonka od verate be torlak

File: user/views.py [hunk 5]
```diff
@@ -27,3 +33,66 @@ class CustomTokenObtainPairView(TokenObtainPairView):
...
+    def retrieve(self, request, pk=None):
+        """
+        Retrieve a specific code snippet by ID.
+        """
+        snippet = get_object_or_404(CodeSnippet, pk=pk)
+        serializer = CodeSnippetSerializer(snippet)
+        return Response(serializer.data)
+
+    def create(self, request, *args, **kwargs):
+
```
Comment (themane04) [inline user/views.py:54]: User file_hash from utils to save the image only once and use it everywhere so you don't have to save the exact same image multiple times

**Pull request merged**